### PR TITLE
Trade Mall (Trade Outpost Alternate)

### PR DESCRIPTION
**Pull request by Cheackraze**
## About the PR

Adds an alternative trade outpost for some round variation. After lots of playtesting and feedback of the original outpost, this one aims to resolve some of the broader player feedback. Mostly of, it being too large and isolating of the areas, and while having lots of blank space to make player spaces is a sound idea, only the large abandoned atrium ever really got used.

Big thanks to mariusdemeier and trucy_deuxy for initial layout and design work.

TODO:
[x] add cameras
[x] modify kitchen dock (it needs 3 more tiles of clearance to fit a bocadillo)
[x] power pass, make sure everything is properly powered (kitchen and atm area had issues)
## Why / Balance

Variety/Feedback
## How to test

Load up a normal (non-debug) round of frontier. There will be a 50/50 chance of it spawning. You may have to `restartroundnow` a few times if it doesnt roll right.
## Media

TradeMall-0

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added the Trade Mall, an alternate to the Trade Outpost.

### CODE REVIEW

**Comment by PeccNeck** [comment]
I've already voiced this once in the Discord forum post, but I'm very iffy about having separate areas for guard and NFSD. They've definitely been nicely decorated, but it ends up feeling redundant to me (NFSD especially, who have their entire POI). They also both share the same basic access, too (Security), so there's nothing stopping the Guard from waltzing over to NFSD and vice versa.

Semi-related, sad to see the NFSD intercom in the Guard section didn't make the cut, as they have no direct channels to NFSD otherwise, but at least there's one in the office on the other side of the hall.

**Comment by Cheackraze** [comment]
this office is actually intended as a shared nfsd/sec guard post, to be manned by either. I mainly just wanted to give them both their own secured docking access, as the reasoning behind the 'split'. Overall i concurr

image


> I've already voiced this once in the Discord forum post, but I'm very iffy about having separate areas for guard and NFSD. They've definitely been nicely decorated, but it ends up feeling redundant to me (NFSD especially, who have their entire POI). They also both share the same basic access, too (Security), so there's nothing stopping the Guard from waltzing over to NFSD and vice versa.
> 
> Semi-related, sad to see the NFSD intercom in the Guard section didn't make the cut, as they have no direct channels to NFSD otherwise, but at least there's one in the office on the other side of the hall.
> 
> And completely separately, I've just noticed that the decals for Cargo Bay 3 overlap the dark tiles on one side and are overlapped by them on the other.

The cargo bay area is mostly untouched from the original designer, but I can go in and alter the decals when I make my final pass

**Comment by blackknight954** [comment]
So overall this just looks like a revamped trade outpost? Why the 1/2 chance to spawn one or the other?

**Comment by Cheackraze** [comment]
> So overall this just looks like a revamped trade outpost? Why the 1/2 chance to spawn one or the other?

for variety? why not?

**Comment by blackknight954** [comment]
> > So overall this just looks like a revamped trade outpost? Why the 1/2 chance to spawn one or the other?
> 
> for variety? why not?

Yeah I guess, but I'm imagining that trade outpost is just gonna remain a ghost town whereas the mall is gonna get all the action, but we'll see.

**Comment by Cheackraze** [comment]
> > > So overall this just looks like a revamped trade outpost? Why the 1/2 chance to spawn one or the other?
> > 
> > 
> > for variety? why not?
> 
> Yeah I guess, but I'm imagining that trade outpost is just gonna remain a ghost town whereas the mall is gonna get all the action, but we'll see.

my intent with the closed up shops is to be more heavy-handed doing the same thing the original is trying to do. So, players making shops and populating it a bit more might carry over to the original trade outpost, which is set up for the same kind of thing but just is a little lighter in its suggestions for it

**Comment by Houtblokje** [comment]
Overall, looks pretty good, I definitely like the shape and most of the layout. Especially the cargo area looks pretty cool, since you can easily reach and interact with the other cargo rooms, which I think was one of the big issues with the current trade station. 

but I had a few questions about this new layout. Mainly:
I see the mail office and the janitorial office here. Does this mean those facilities will be removed from Frontier outpost? Will their jobs be located on Trade? if so, is there a console to re-open the slots?
If this station is going to have a little medical area, maybe it could also get a cryo area, to take the pressure off frontier outpost during the end of game rush and to lure people more to it?

Not sure how I feel about the big open area, it often goes unused since nobody knows its allowed to be modified and what modifications are allowed, but I guess it is nice to have a big space to do fun stuff in.

The STC room could do with a water cooler.

**Comment by Cerebral-Error** [comment]
> Overall, looks pretty good, I definitely like the shape and most of the layout. Especially the cargo area looks pretty cool, since you can easily reach and interact with the other cargo rooms, which I think was one of the big issues with the current trade station.
> 
> but I had a few questions about this new layout. Mainly: I see the mail office and the janitorial office here. Does this mean those facilities will be removed from Frontier outpost? Will their jobs be located on Trade? if so, is there a console to re-open the slots? If this station is going to have a little medical area, maybe it could also get a cryo area, to take the pressure off frontier outpost during the end of game rush and to lure people more to it?
> 
> Not sure how I feel about the big open area, it often goes unused since nobody knows its allowed to be modified and what modifications are allowed, but I guess it is nice to have a big space to do fun stuff in.
> 
> The STC room could do with a water cooler.

There does look like there's a small Cryo area up near the North end of the station, right next to the toilets. There's a pair of cryo chambers.

**Comment by Houtblokje** [comment]
> > Overall, looks pretty good, I definitely like the shape and most of the layout. Especially the cargo area looks pretty cool, since you can easily reach and interact with the other cargo rooms, which I think was one of the big issues with the current trade station.
> > but I had a few questions about this new layout. Mainly: I see the mail office and the janitorial office here. Does this mean those facilities will be removed from Frontier outpost? Will their jobs be located on Trade? if so, is there a console to re-open the slots? If this station is going to have a little medical area, maybe it could also get a cryo area, to take the pressure off frontier outpost during the end of game rush and to lure people more to it?
> > Not sure how I feel about the big open area, it often goes unused since nobody knows its allowed to be modified and what modifications are allowed, but I guess it is nice to have a big space to do fun stuff in.
> > The STC room could do with a water cooler.
> 
> There does look like there's a small Cryo area up near the North end of the station, right next to the toilets. There's a pair of cryo chambers.

Ah I see it. I expected it near medical however, so maybe something similar can be created instead of the toilets across from medical?

**Comment by Cheackraze** [comment]
> > > Overall, looks pretty good, I definitely like the shape and most of the layout. Especially the cargo area looks pretty cool, since you can easily reach and interact with the other cargo rooms, which I think was one of the big issues with the current trade station.
> > > but I had a few questions about this new layout. Mainly: I see the mail office and the janitorial office here. Does this mean those facilities will be removed from Frontier outpost? Will their jobs be located on Trade? if so, is there a console to re-open the slots? If this station is going to have a little medical area, maybe it could also get a cryo area, to take the pressure off frontier outpost during the end of game rush and to lure people more to it?
> > > Not sure how I feel about the big open area, it often goes unused since nobody knows its allowed to be modified and what modifications are allowed, but I guess it is nice to have a big space to do fun stuff in.
> > > The STC room could do with a water cooler.
> > 
> > 
> > There does look like there's a small Cryo area up near the North end of the station, right next to the toilets. There's a pair of cryo chambers.
> 
> Ah I see it. I expected it near medical however, so maybe something similar can be created instead of the toilets across from medical?

the intent is to make people walk through the station to get to it. Same with having the ATM/shipyard console on the opposite end of the Cargo console. It may 'feel like a pain in the ass' but the intent is that you should walk all through the station with the goal of players having chance interactions in the hallways etc.

As for the janitor/mail, we dont plan on removing them from frontier outpost, its more just another space that they can choose to use instead.

**Comment by Houtblokje** [comment]
> the intent is to make people walk through the station to get to it. Same with having the ATM/shipyard console on the opposite end of the Cargo console. It may 'feel like a pain in the ass' but the intent is that you should walk all through the station with the goal of players having chance interactions in the hallways etc.
> 
> As for the janitor/mail, we dont plan on removing them from frontier outpost, its more just another space that they can choose to use instead.

Not sure what roleplay opporunities you see in delaying people who are actively trying to not play the game anymore, but sure I guess? I do get the other stuff though.

**Comment by PeccNeck** [comment]
Went ahead and submitted some bits and tweaks on #2373, take a look when you can.

**Comment by dvir001** [comment]
Fixed warp point and metadata.

**Comment by Cheackraze** [comment]
this is all good to go now. I cant get an updated map render because my maprenderer breaks on "entity uid 0" internal exception. No substantial changes from the above images, besides a slightly modified kitchen dock and added cameras.

**Comment by arimah** [comment]
Failures come from these lines:

https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L24873-L24875
https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L25349-L25351

tl;dr: don't use a network configurator or multitool's overlay. :D These will have to be edited out manually.

I will give this a proper review in the coming days.

**Comment by arimah** [comment]
I will say this right away, actually: the Christmas trees _have to_ go until we've fixed the present pool. It's _way_ too easy to get gifts that are dangerous or game-breaking in some way.

**Comment by dvir001** [comment]
Fixed the yml, I hope.

**Comment by arimah** [comment]
@dvir001 You don't have to remove `parent: invalid` getting the grid's Transform component. The map deserialiser explicitly ignores that one case, and it'll just be reintroduced whether the map is edited anyway. :>

**Comment by arimah** [comment]
> I'm not convinced your reasoning is correct here. My latest updates changed exactly 0 network devices and i hadn't used any kind of network config tool for this.

🤔 You're right, it's actually not the overlay that causes `configurators` to be set. Merely adding a device to a multitool or network config is enough to put it in this list. TIL!

**Comment by Cheackraze** [comment]
> I will say this right away, actually: the Christmas trees _have to_ go until we've fixed the present pool. It's _way_ too easy to get gifts that are dangerous or game-breaking in some way.

The gift pool has literally nothing dangerous in it, it was removed by wizden last Christmas after they kept giving out thronglers on it. If anything, you can lower the chance for the 'insane' version, but this is kinda the whole meaning and draw of the Christmas gifts

**Comment by arimah** [comment]
> The gift pool has literally nothing dangerous in it, it was removed by wizden last Christmas after they kept giving out thronglers on it.

Just a few weeks ago, Wookie spawned a gift-giving Christmas tree that someone got a Syndicate monkey box from. It made a space bear on FO. Feel free to check the ahelp relay. In my own testing, I've managed to unwrap a syndicate footsoldier, a holoparasite, a Syndicate hardbomb, an anomaly, and more. Whatever is supposed to have been removed, I don't know, but the insane gifts are still, well, insane.

**Comment by PeccNeck** [comment]
> Failures come from these lines:
> 
> https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L24873-L24875
> 
> https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L25349-L25351
> 
> tl;dr: don't use a network configurator or multitool's overlay. :D These will have to be edited out manually.

Oops... Those were me. Didn't know! Is there a better way to link devices for any future mapping I do, or should I just be more vigilant and edit these out in post?

**Comment by Cheackraze** [comment]
its gotta be something specific because I linked the entire outpost up to air alarms, fire locks, shutters, the whole shebang and it didnt fail like this

> > Failures come from these lines:
> > https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L24873-L24875
> > 
> > https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L25349-L25351
> > 
> > tl;dr: don't use a network configurator or multitool's overlay. :D These will have to be edited out manually.
> 
> Oops... Those were me. Didn't know! Is there a better way to link devices for any future mapping I do, or should I just be more vigilant and edit these out in post?

**Comment by PeccNeck** [comment]
> its gotta be something specific because I linked the entire outpost up to air alarms, fire locks, shutters, the whole shebang and it didnt fail like this
> 
> > > Failures come from these lines:
> > > https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L24873-L24875
> > > 
> > > https://github.com/new-frontiers-14/frontier-station-14/blob/ea05405aefeb479a61a88c7f7aa55905f916d1ab/Resources/Maps/_NF/POI/trademall.yml#L25349-L25351
> > > 
> > > tl;dr: don't use a network configurator or multitool's overlay. :D These will have to be edited out manually.
> > 
> > 
> > Oops... Those were me. Didn't know! Is there a better way to link devices for any future mapping I do, or should I just be more vigilant and edit these out in post?

Huh. Definitely odd... Maybe it is the overlay after all? I do tend to use that just to be sure I haven't missed anything (and sometimes I have).

Either way, those lines are definitely from my PR according to the git blame; clearly I did something wrong, just a matter of figuring out what. Had this happen to me with other grids I've saved as well, so whatever it is I'm doing, I'm doing it consistently.

**Review by arimah — CHANGES_REQUESTED** [review]
Overall, I like it. It has a very unique shape and looks good in many, many ways. We've discussed a few of the larger initiatives in VC, none of which are blockers for getting this merged. The following is therefore largely a list of nitpicks. :)

image
This corner is dark. Suggested new light placement:
image

The corridor outside also feels a touch underlit compared to the rest:
image

image
Another dark place. I suggest putting a light in the middle of that wall, symmetric with the shop opposite.

image
The only food vendor in the entire system, wow! ... I'm not sure we should have it there. :)

image
North bus dock is also pretty dark. Not a huge deal.

image
As per our discussion in VC, replace the ChefVend with a sink and replace the range with a deep fryer.

image
The windoors here are almost invisible. Suggest removing them, it's just a kitchen counter. :)

image
This station map has no power cell. Swap it for the `[Powered]` version.

image
This station map too.
Also, catwalk under the generators would make it instantly obvious that there's HV power cable.

image
Left-hand conveyor belt needs to be turned 180° – it moves in the opposite direction to the other two. :) Also the recycler and mat reclaimer do the same thing now, we can skip one of them.

And lastly, I think it could do with a few dirt decals. All the dirty spessmen feet tracking mud everywhere must surely give the janitor something to clean!

**Comment by Cheackraze** [comment]
all concerns listed addressed, this should be good to merge!

**Comment by whatston3** [comment]
A dock numbering scheme might be handy.  Thoughts on the one below?  Similar to the Trade Outpost names:

image

**Comment by Cheackraze** [comment]
looks fine to me, the only reason I havent been labeling the docks is localization issues but that would work fine if you want to commit or make changes

**Comment by dvir001** [comment]
> looks fine to me, the only reason I havent been labeling the docks is localization issues but that would work fine if you want to commit or make changes

They actually should support FTL format if you use FTL in the name as far as I know, I might have to retest.

**Comment by whatston3** [comment]
> > looks fine to me, the only reason I havent been labeling the docks is localization issues but that would work fine if you want to commit or make changes
> 
> They actually should support FTL format if you use FTL in the name as far as I know, I might have to retest.

They don't right now, I got lazy, but it's an easy addition.  One sec.

File: Resources/Prototypes/_NF/PointsOfInterest/trademall.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Basic Trade Outpost buying point for economy and great wealth
+- type: pointOfInterest
+  id: TradeMall
+  name: Trade Mall
+  rangeMin: 1500
+  rangeMax: 2500
+  IFFColor: "#37C837"
+  spawnGroup: MarketStation
+  gridPath: /Maps/_NF/POI/trademall.yml
+  busStop: true
```
Comment (dvir001) [inline Resources/Prototypes/_NF/PointsOfInterest/trademall.yml:19]: Note some of the changes on this yml are not in the comp anymore, need to update

**Comment by whatston3** [comment]
> looks fine to me, the only reason I havent been labeling the docks is localization issues but that would work fine if you want to commit or make changes

https://github.com/new-frontiers-14/frontier-station-14/pull/2467
boop

**Comment by whatston3** [comment]
> are both ready to go? or do we need to now update all the dock labels too again

Whichever one merges first, the other should be updated.

Also, fixed up the trade mall prototype re: dvir's comments (should've done that earlier, my bad).

**Comment by whatston3** [comment]
Dusty's requested changes are up here - three airlocks per cargo dock (all labelled), swaps the position of the shelf and westmost light in the kitchen: https://github.com/new-frontiers-14/frontier-station-14/commit/26e8eef6bc767c24ea524f512ca139c3364024b2

All of the new cargo airlocks should have techmaint floors underneath them with the same warning decals as the existing ones.

Cargo bay 3, three red airlocks and docking airlocks.
image

Kitchen, note the swapped shelf and wall-mounted light.
image

**Review by arimah — APPROVED** [review]
image
